addons/web/static/tests/views/list_view_tests.js

```diff
@@ -18299,4 +18299,32 @@ QUnit.module("Views", (hooks) => {
         await click(document, "body .modal footer button.btn-primary");
         assert.verifySteps(["a"]);
     });
+
+    QUnit.test("x2many onchange, check result", async function (assert) {
+        assert.expect(2);
+
+        serverData.models.foo.onchanges = {
+            m2m: function () {},
+        };
+
+        await makeView({
+            type: "list",
+            resModel: "foo",
+            serverData,
+            arch: `<tree editable="bottom" expand="1">
+                    <field name="m2m" widget="many2many_tags"/>
+                    <field name="m2o"/>
+                </tree>`,
+            mockRPC(route, args) {
+                if (args.method === "onchange") {
+                    return {"value": {"m2o": [3, "Value 3"]}};
+                }
+            },
+        });
+
+        assert.equal(target.querySelector(".o_data_row .o_list_many2one").dataset.tooltip, "Value 1");
```

Comment:
don't assert the tooltip but the content
strictEqual rather than equal

same  for last assert